content/en/docs/concepts/architecture/cgroups.md

```diff
@@ -0,0 +1,134 @@
+  * [containerd](https://containerd.io/) since 1.4
+  * [cri-o](https://cri-o.io/) since 1.20
+* Kubelet and container runtime are configured to use the [systemd cgroup driver](/docs/setup/production-environment/container-runtimes#systemd-cgroup-driver)
+
+### Linux Distribution cgroupv2 support
+
+Many Linux Distributions have already switched over to use cgroupv2 by default, for example:
+
+<!-- the list should be kept in sync with https://github.com/opencontainers/runc/blob/main/docs/cgroup-v2.md -->
+* Container Optimized OS M97
+* Ubuntu (since 21.10, 22.04+ recommended)
+* Debian GNU/Linux (since Debian 11 buster)
+* Fedora (since 31)
+* Arch Linux (since April 2021)
+* RHEL and RHEL-like distributions (since 9)
+
+To check if your distribution is using cgroupv2, refer to your distribution's
+documentation or follow the steps [below](#check-cgroup-version) to verify the
+configuration.
+
+You can also enable cgroupv2 manually on your Linux distribution by modifying
+the kernel boot arguments in the GRUB command line, and setting
+`systemd.unified_cgroup_hierarchy=1`, however it's recommended to use a
+distribution that already enables cgroupv2 by default.
+
+
+### Migrating to cgroupv2
+
+To migrate to cgroupv2, update to a newer kernel version that enables cgroupv2
+by default, ensure your container runtime supports cgroupv2, and configure
+kubelet and container runtime are configured to use the [systemd cgroup
+driver](/docs/setup/production-environment/container-runtimes#systemd-cgroup-driver).
+
+Kubelet will automatically detect that the OS is running on cgroupv2 and will
+perform accordingly, no additional configuration is required.
+
+There should not be any noticeable difference in the user experience when
+switching to cgroup v2, unless users are accessing the cgroup file system
+directly, either on the node or from within the containers.
+
+Cgroup V2 uses a new API as compared to cgroup V1, so if there are any
+applications that directly access the cgroup file system, they need to be
+updated to newer versions that support cgroupv2. For example:
+
+* Some third party monitoring and security agents may be dependent on cgroup filesystem.
+ Update them to the latest versions that support cgroupv2
+* If you are running [cAdvisor](https://github.com/google/cadvisor) as a
+ daemonset for monitoring pods and containers, update it to latest version (v0.45.0)
+* If you use JDK (Java workload), prefer to use JDK 11.0.16 and later or JDK 15
+ and later, which [fully support
+ cgroupv2](https://bugs.openjdk.org/browse/JDK-8230305)
+
+
+## Identifying cgroup version used on Linux Nodes  {#check-cgroup-version}
+
+The cgroup version is dependent on the Linux distribution being used and the
+default cgroup version configured on the OS. To check which cgroup version your
+OS Distro is using, you can run the `stat -fc %T /sys/fs/cgroup/` command on
+the node and check if the output is `cgroup2fs`:
+
+```shell
```

Comment:
Separate the command code block from the output code block

<command block>

For cgroup v2, the output is `cgroup2fs`. For cgroup v1, the output is `tmpfs`.

Reply:
Done, let me know what you think.

Reply:
LGTM but remove the `$`

Reply:
Done